### ISSUE TITLE
From text file

### ISSUE THREAD

**Issue by cannyboy** [cparnot/ASCIImage#5]
Might be useful, if same image needs to be used in several source files.

```
+ (PARImage *)imageWithASCIITextFile:(NSString *)textFile color:(PARColor *)color shouldAntialias:(BOOL)shouldAntialias {
    NSString *type = @"txt";
    NSString *separation = @"\n";
    NSString *fileText = [NSString stringWithContentsOfFile:[[NSBundle mainBundle] pathForResource:textFile ofType:type] encoding:NSUTF8StringEncoding error:nil];
    NSArray *rep = [[NSArray alloc] initWithArray:[fileText componentsSeparatedByString:separation]];
    return [self imageWithASCIIRepresentation:rep contextHandler:[self contextHandlerForImageWithColor:color shouldAntialias:shouldAntialias]];
}
```

![asciitextfile](https://cloud.githubusercontent.com/assets/1590436/6845460/e222c902-d3ac-11e4-8902-0835aec9379b.png)

**Comment by cparnot** [comment]
I think any I/O should stay out of such a library. In general, this is probably true of any library :-)

Also, limiting yourself to a file path in the current bundle seems... a bit limiting. For paths out of the bundle, then another method is needed.

And then, one of the main idea of ASCIImage was for me to have the image strings right there with the code instead of having to chase down a file in the resources.

Finally, I am always wary of multiplying the convenience methods that only save a couple of lines of code and that can lead to API confusion and multiplication. I realize you only propose one here, but I guess I am worried about the slippery slope. I am already not so happy about having 2 separate methods when actually the 'simple' one can be expressed very simply with the block method. The 'simple' method is only there because that was initially the only method I had.

In conclusion, I'll close the issue, sorry!

To end on a positive note, however, I'd be happy to allow input to be a string instead of an array, using (id) as the type so both can work. It's very easy to handle both by checking the class and split a string into multiple lines as part of preprocessing.
